src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -287,6 +287,17 @@ def _remove_comments_from_json(template, preserve_order=True, file_path=None):
         raise CLIError("Failed to parse the JSON data, please check whether it is a valid JSON format")
 
 
+def _remove_multiline_from_json(template):
+    result = template
+    # find all text within bock quotes (single-line and multiline)
+    for match in re.finditer(r'((?<!\\)")((?:.?(?!(?<!\\)"))*.?)??"', template, re.DOTALL):
```

Comment:
In order to increase the readability of the code, could you please add some comments explaining the regular expression in detail? For example, add some matching samples to it?